examples/unions/unions.bal

```diff
@@ -19,14 +19,73 @@ public function main() {
 
     io:println(nameToString(name1));
     io:println(nameToString(name2));
+
+    map<string?> grades1 = {
+        math: "80",
+        physics: (),
+        chemistry: "76"
+    };
+    // Parsing a map with grades values that are either the string
+    // representation of an integer or nil results in a map of
+    // the just the non-nil grades as integers.
+    map<int>|error parseGrades1 = parseGrades(grades1);
+    io:println(parseGrades1);
+
+    map<string> grades2 = {
+        math: "80",
+        physics: "N/A",
+        chemistry: "76"
+    };
+    // Attempting to parse a map with string values that
+    // are not the string representation of a number results
+    // in the function terminating early, returning the error.
+    map<int>|error parseGrades2 = parseGrades(grades2);
+    io:println(parseGrades2);
 }
 
 function nameToString(Name nm) returns string {
     // Checks whether `nm` belongs to `string` type.
     if nm is string {
-
+        // The type of `nm` is narrowed to `string` here.
+        // Therefore, you can directly return `nm` from this
+        // function that specifies `string` as the return type.
         return nm;
     } else {
```

Comment:
We can remove this else block too

Reply:
Kept this intentionally to incrementally introduce narrowing and then narrowing based on reachability.